### ISSUE TITLE
Requesting multiple directions to display a route

### ISSUE THREAD

**Issue by BarmaleyHD** [mapbox/mapbox-gl-js#6710]
Hello everyone

I'm working on map that displays already optimized routes with multiple stops. 
Optimization API from MapBox doesn't take in account time windows of arrival so I can't use it. 
![selectedroute](https://user-images.githubusercontent.com/16465097/40374602-9fe4bd7a-5da6-11e8-8cfd-847116b30eb0.png)

Right now I'm requesting direction from Start to Stop1 , Stop1 to Stop2 and so on.
I believe it's not very efficient and in future when route could have more than 60 stops it will hit request limitation. 
Basically in this particular situations I just need to draw a line that represents a route and get information about it length and time. I don't need turn-by-turn information. 

Would be great if I could request GeoJSON information about whole route in 1 request. 

Thank you .

**Comment by jfirebaugh** [comment]
Hi, thanks for using Mapbox. This issue tracker is for reporting bugs and feature requests in Mapbox GL, the map renderer, while your request seems to be about the Mapbox Directions API. For that topic, please refer to our [help documentation](https://www.mapbox.com/help/), and if you can't find the answer there, [contact support](https://www.mapbox.com/contact/). Thanks!

**Comment by myselfsum** [comment]
@BarmaleyHD, could you please tell me how do you are sending multiple direction request using MapBox due to more than 25 waypoints. I don't know how to send.

I am integrating MapBox Direction API in react. I have more than 25 waypoints (around 500). How do I use all the waypoints using "mapbox-gl-directions". Because, I can't send not more than 25 wp. 

Below is my sample code:

import MapboxDirections from '@mapbox/mapbox-gl-directions/dist/mapbox-gl-directions'

componentDidMount() {
	let directions = new MapboxDirections({
	            accessToken: mapboxgl.accessToken,
	            unit: 'metric',
	            profile: 'mapbox/driving',
	            interactive: false,
	            controls: false
	        });

	this.map = new mapboxgl.Map({
            container: 'map',
            style: 'mapbox://styles/mapbox/streets-v11',
            center: [3.1026, 36.6686],
            //center: [this.state.lng, this.state.lat],
            zoom: 10
        });

	this.map.on('load', function() {
	    directions.setOrigin([-117.1425, 32.63638889]);
	    directions.addWaypoint(0, [-117.1425, 32.63638889]);
	    directions.addWaypoint(1, [-117.195, 32.75416667]);
	    ---
	    ---
	    directions.addWaypoint(23, [-116.5012667, 32.92583333]);
	    directions.setDestination([-116.5616667, 32.93583333]);
	})
	this.map.addControl(directions, 'top-left');
}

Any suggestion? Thanks in advance.
